Sources/MapboxNavigation/NavigationViewData.swift

```diff
@@ -0,0 +1,28 @@
+
+import UIKit
+import MapboxCoreNavigation
+
+protocol NavigationViewData: class {
+    var navigationView: NavigationView! { get }
+    var navigationService: NavigationService! { get }
+    var navigationViewController: UIViewController! { get }
+}
+
+
+protocol ViewCycleObserver {
+    func onViewDidLoad(_: UIView)
+    func onViewWillAppear(_: Bool)
+    func onViewDidAppear(_: Bool)
+    func onViewWillDisappear(_: Bool)
+    func onViewDidDisappear(_: Bool)
+    func onDidLayoutSubviews()
```

Comment:
The “on-” naming convention is atypical of Swift code. Instead, it should be sufficient to name the methods, e.g., `viewDidLoad`. If the problem is a conflict with the usual UIView methods, then we should make the method names more specific.

Overall, it seems like ViewCycleObserver avoids creating a separate protocol for each view for which another class might want to track. Separate protocols would allow a single controller class to track the lifecycle of multiple views without resorting to a series of `if let`s. But if we stick to the ViewCycleObserver approach, then let’s call it a delegate protocol. It looks like each of the classes that currently conform to ViewCycleObserver is a NavigationComponent, so what if we formalize that relationship by calling it NavigationComponentDelegate?

Reply:
I can hardly imagine a case where one controller will have to maintain multiple views, so I've decided to stick to this delegate protocol solution.